clients/linux/src/filetree.rs

```diff
@@ -183,38 +192,49 @@ impl FileTree {
                         parent = model.get_iter(&path).unwrap();
                     }
                     _ => {
-                        if tree_iter_value!(model, &parent, 3, String) == format!("{:?}", FileType::Document)  {
+                        if tree_iter_value!(model, &parent, 3, String) == format!("{:?}", FileType::Document) {
                             path.up();
                             parent = model.get_iter(&path).unwrap();
                         }
                     }
                 }
 
-                if tree_iter_value!(model, &parent, 3, String) == format!("{:?}", FileType::Document)  {
+                if tree_iter_value!(model, &parent, 3, String) == format!("{:?}", FileType::Document) {
                     path.up();
                     parent = model.get_iter(&path).unwrap();
                 }
 
-                let (paths, _) = w.get_selection().get_selected_rows();
+                match info {
+                    LOCKBOOK_FILES_TARGET_INFO => {
+                        let (paths, _) = w.get_selection().get_selected_rows();
 
-                let iters = paths.iter().map(|selected| model.get_iter(selected).unwrap()).collect::<Vec<TreeIter>>();
-                let ids = iters.iter().map(|iter| Uuid::parse_str(&tree_iter_value!(model, iter, 2, String)).unwrap()).collect::<Vec<Uuid>>();
+                        let iters = paths.iter().map(|selected| model.get_iter(selected).unwrap()).collect::<Vec<TreeIter>>();
+                        let ids = iters.iter().map(|iter| Uuid::parse_str(&tree_iter_value!(model, iter, 2, String)).unwrap()).collect::<Vec<Uuid>>();
 
-                let parent_id = Uuid::parse_str(tree_iter_value!(model, &parent, 2, String).as_str()).unwrap();
+                        let parent_id = Uuid::parse_str(tree_iter_value!(model, &parent, 2, String).as_str()).unwrap();
 
-                ids.iter().enumerate().for_each(|(index, id)| {
-                    match c.move_file(id, parent_id) {
-                        Ok(_) => {
-                            Self::move_iter(&model, &iters[index], &parent, true);
-                            model.remove(&iters[index]);
-                        }
-                        Err(err) => m.send_err_dialog("moving", err)
+                        ids.iter().enumerate().for_each(|(index, id)| {
+                            match c.move_file(id, parent_id) {
+                                Ok(_) => {
+                                    Self::move_iter(&model, &iters[index], &parent, true);
+                                    model.remove(&iters[index]);
+                                }
+                                Err(err) => m.send_err_dialog("moving", err)
+                            }
+                        });
                     }
-                });
+                    URI_TARGET_INFO => {
+                        let parent_id = Uuid::parse_str(tree_iter_value!(model, &parent, 2, String).as_str()).unwrap();
```

Comment:
`parent_id` is the exact same variable in both branches of this match statement, correct? Maybe it can be declared once outside the match.